fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -423,8 +423,8 @@ public void dropDb(String dbName, boolean isForceDrop) throws DdlException, Meta
                 }
 
                 // save table names for recycling
-                Set<String> tableNames = db.getTableNamesWithLock();
-                runnableList = unprotectDropDb(db, isForceDrop, false);
+                Set<String> tableNames = new HashSet<String>();
+                runnableList = unprotectDropDb(db, isForceDrop, false, tableNames);
```

Comment:
You can copy the name to a new set here, instead of add a param in unprotectDropDb.

Reply:
Besides, better to change the getTableNamesWithLock to a name that reminds the caller that this is just a view of the tablet map.

Reply:
ok,i have fixed it.